### PR TITLE
feat: create Translator role

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1779
https://oat-sa.atlassian.net/browse/ADF-1780
https://oat-sa.atlassian.net/browse/ADF-1781
https://oat-sa.atlassian.net/browse/ADF-1783
https://oat-sa.atlassian.net/browse/HKD-6

# Goal 

Introduce new translator role that will be able to perform translations actions

# PRs
* https://github.com/oat-sa/tao-core/pull/4094
* https://github.com/oat-sa/extension-tao-test/pull/476
* https://github.com/oat-sa/extension-tao-itemqti/pull/2571
* https://github.com/oat-sa/extension-tao-testqti/pull/2503

# TODO 

- [ ] Update composer dependencies

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -61,7 +61,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis": ">=15.36.1",
-        "oat-sa/tao-core": ">=53.0.0",
+        "oat-sa/tao-core": "dev-feat/ADF-1781/translations-feature",
```
Comment (gabrielfs7) [inline composer.json:64]: # TODO 

- [ ] Update composer dependencies

**Review by shpran — APPROVED** [review]
Good job! 👍🏻

File: models/classes/Translation/Listener/ItemCreatedEventListener.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        FeatureFlagCheckerInterface $featureFlagChecker,
+        Ontology $ontology,
+        LoggerInterface $logger
+    ) {
+        $this->featureFlagChecker = $featureFlagChecker;
+        $this->ontology = $ontology;
+        $this->logger = $logger;
+    }
+
+    public function populateTranslationProperties(Event $event): void
```
Comment (gabrielfs7) [inline models/classes/Translation/Listener/ItemCreatedEventListener.php:48]: If this classes is called `ItemCreatedEventListener`, should not we restrict the event to instance of `ItemCreatedEvent` only?

@shpran 

```suggestion
    public function populateTranslationProperties(ItemCreatedEvent $event): void
```

File: models/classes/Translation/Listener/ItemCreatedEventListener.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+namespace oat\taoItems\model\Translation\Listener;
+
+use oat\generis\model\data\Ontology;
+use oat\oatbox\event\Event;
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\tao\model\TaoOntology;
+use oat\taoItems\model\event\ItemCreatedEvent;
+use Psr\Log\LoggerInterface;
+
+class ItemCreatedEventListener
```
Comment (gabrielfs7) [inline models/classes/Translation/Listener/ItemCreatedEventListener.php:32]: - [ ] @shpran It is missing tests, please also check the other new classes. I am not sure all them have tests. Thank you!

File: models/classes/Translation/Listener/ItemCreatedEventListener.php (hunk 2 above)
Comment (shpran) [reply models/classes/Translation/Listener/ItemCreatedEventListener.php]: Done ✅

File: models/classes/Translation/Listener/ItemCreatedEventListener.php (hunk 3 above)
Comment (shpran) [reply models/classes/Translation/Listener/ItemCreatedEventListener.php:31]: Done ✅

**Pull request merged**